src/server/login.js

```diff
@@ -55,18 +55,50 @@ module.exports = function (client, server, options) {
   }
 
   function onEncryptionKeyResponse (packet) {
+    const mcData = require('minecraft-data')(client.version)
+
+    let packetVerifyToken
+    let signature
+
+    if (mcData.supportFeature('signatureEncryption')) {
+      // 1.19+
+      if (packet.hasVerifyToken) {
+        packetVerifyToken = packet.crypto.verifyToken
+      } else {
+        signature = packet.crypto
+      }
+    } else {
```

Comment:
I think it would be better to avoid else here and just have a if signatureNotEncrypted
It's not that there was no feature before, it's that there was a different feature

Reply:
Optional though, not absolutely needed to do